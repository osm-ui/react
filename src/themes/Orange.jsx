import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from 'themes/colors';
import { buildDarkThemeConfig } from 'helpers/themes';
import { config as defaultThemeConfig, StyledSpan } from './Default';

export const config = _.merge(
  {},
  buildDarkThemeConfig(defaultThemeConfig, colors, 'orange'),
  {
    form: {
      button: {
        warning: {
          color: colors.white,
          backgroundColor: colors.orange1,
          borderColor: colors.orange1,
          hoverBackgroundColor: colors.orange3,
          hoverBorderColor: colors.orange3,
          focusBackgroundColor: colors.orange3,
          focusBorderColor: colors.orange3,
          activeBackgroundColor: colors.orange3,
          activeBorderColor: colors.orange3
        }
      }
    }
  }
);

const OrangeTheme = ({ children, ...rest }) => (
  <ThemeProvider theme={config}>
    <StyledSpan>
      {React.Children.map(children, child => {
        if (React.isValidElement(child)) {
          return React.cloneElement(child, rest);
        }

        return child;
      })}
    </StyledSpan>
  </ThemeProvider>
);

OrangeTheme.propTypes = {
  children: PropTypes.node.isRequired
};

OrangeTheme.displayName = 'OrangeTheme';

export default OrangeTheme;
