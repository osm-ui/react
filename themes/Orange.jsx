
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from '../colors';
import { buildDarkThemeConfig } from '../helpers/themes';
import {
    config as defaultThemeConfig,
    StyledDiv,
} from './Default';


export const config = _.merge(
    {},
    buildDarkThemeConfig(defaultThemeConfig, colors, 'orange'),
    {
        form: {
            button: {
                warning: {
                    color: colors.white,
                    backgroundColor: colors.orange1,
                    borderColor: colors.orange1,
                    hoverBackgroundColor: colors.orange3,
                    hoverBorderColor: colors.orange3,
                    focusBackgroundColor: colors.orange3,
                    focusBorderColor: colors.orange3,
                    activeBackgroundColor: colors.orange3,
                    activeBorderColor: colors.orange3,
                },
            },
        },
    },
);


const RedTheme = ({ children, ...rest }) => (
    <ThemeProvider theme={config}>
        <StyledDiv {...rest}>
            {children}
        </StyledDiv>
    </ThemeProvider>
);


RedTheme.propTypes = {
    children: PropTypes.node.isRequired,
};

RedTheme.displayName = 'RedTheme';

export default RedTheme;
