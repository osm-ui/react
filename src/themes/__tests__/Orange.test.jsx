import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from 'helpers/tests';
import Orange from 'themes/Orange';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Orange));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Orange));
});
