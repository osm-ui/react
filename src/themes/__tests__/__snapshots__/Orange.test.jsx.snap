// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<Default__StyledSpan>
  <OrangeTheme
    key=".0"
  >
    This is a text
  </OrangeTheme>
</Default__StyledSpan>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<Default__StyledSpan>
  <OrangeTheme
    key=".0"
  >
    <div />
  </OrangeTheme>
</Default__StyledSpan>
`;
